keps/sig-node/3721-support-for-env-files/README.md

```diff
@@ -0,0 +1,552 @@
+|8. The specified file contains invalid env var names. | Pod created | Container fails to start and erorr message is reported in the events.|
+
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+No
+
+##### Unit tests
+
+- [Pod Validation
+  tests](https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/validation/validation_test.go)
+- TODO
+
+##### Integration tests
+
+- Pods tests: https://github.com/kubernetes/kubernetes/blob/91ee30074bee617d52fc24dc85132fe948aa5153/test/integration/pods/pods_test.go
+
+##### e2e tests
+
+The tests will be guarded by the `[Feature:EnvFiles]` tag.
+- Pods e2e test: https://github.com/kubernetes/kubernetes/blob/91ee30074bee617d52fc24dc85132fe948aa5153/test/e2e/common/node/pods.go
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind a feature flag.
+- unit and integration tests completed.
+- Initial e2e tests completed and enabled.
+
+#### Beta
+
+- Feature gate is enabled by default.
+- Allowing time for feedback.
+
+#### GA
+
+- TBD
+
+### Upgrade / Downgrade Strategy
+
+kube-apiserver should be updated before kubelets in that order. Upgrade involves
+draining all pods from the node, update to a matching kubelet and making the
+node schedulable again. Downgrade involves doing the above in reverse.
+
+### Version Skew Strategy
+
+Since kubelet versions must not be ahead of kube-apiserver versions, the older
+versions of kubelet or if the feature gate is disabled in kubelet, the fields
+will be ignored.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: EnvFiles
+  - Components depending on the feature gate:
+    - kube-apiserver
+    - kubelet
+
+###### Does enabling the feature change any default behavior?
+
+No
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, Containers already running will have their environment variables populated
+from the specified file. But on restart, the fields will be stripped and the
+feature will no longer work.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Newly started or restarted containers will have their environment variables
+populated from the specified file.
+
+###### Are there any tests for feature enablement/disablement?
+
+No
+
+### Rollout, Upgrade and Rollback Planning
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+The [Version Skew Strategy](#version-skew-strategy) section covers this point.
+It will not impact any already running workloads.
```

Comment:
Can we drop the "It will not impact any already running workloads." phrase here?  The link to the skew strategy is great.  But any statements that suggest safety of rollback or roll forward after a rollback might be counter productive, since changing the environment variables provided to a workload could have significant negative impacts to it..